python-package/xgboost/dask.py

```diff
@@ -805,12 +837,43 @@ def _dmatrix_from_list_of_parts(
     return _create_dmatrix(**kwargs)
 
 
-async def _get_rabit_args(n_workers: int, client: "distributed.Client") -> List[bytes]:
-    '''Get rabit context arguments from data distribution in DaskDMatrix.'''
-    env = await client.run_on_scheduler(_start_tracker, n_workers)
+async def _get_rabit_args(
+    n_workers: int, dconfig: Optional[Dict[str, Any]], client: "distributed.Client"
+) -> List[bytes]:
+    """Get rabit context arguments from data distribution in DaskDMatrix.
+
+    """
+    # There are 3 possible different addresses:
+    # 1. Provided by user via dask.config
+    # 2. Guessed by xgboost `get_host_ip` function
+    # 3. From dask scheduler
+    # We try 1 and 3 if 1 is available, otherwise 2 and 3.
+    valid_config = ["scheduler_address"]
+    # See if user config is available
+    if dconfig is not None:
+        for k in dconfig:
+            if k not in valid_config:
+                raise ValueError(f"Unknown configuration: {k}")
+        host_ip: Optional[str] = dconfig.get("scheduler_address", None)
```

Comment:
I'm curious why this `dconfig` object is being passed around instead of accessing `dask.config.get("xgboost.scheduler_address")` directly.

Reply:
I couldn't access it on the worker side for some reasons.  I read the doc from https://docs.dask.org/en/stable/configuration.html , it's not quite clear to me how to make sure everything is synced.

Reply:
I also read this section https://docs.dask.org/en/stable/configuration.html#downstream-libraries .  In XGBoost dask is loaded lazily so having dask in init might be less desirable due to the size of dependency.  So I went on and read this:


> However, downstream libraries may choose alternative solutions, such as isolating their configuration within their library, rather than using the global dask.config system. All functions in the dask.config module also work with parameters, and do not need to mutate global state.

So far using this parameter passing approach seems to be easiest way to get 1 parameter in.

Reply:
That makes sense. Generally the Dask config should be available on the scheduler/workers but that can be deployment dependent.